controllers/servers/host_definer/watcher/csi_node_watcher.py

```diff
@@ -0,0 +1,86 @@
+
+
+class CsiNodeWatcher(Watcher):
+
+    def add_initial_csi_nodes(self):
+        csi_nodes = self._get_csi_nodes_with_driver()
+        for csi_node in csi_nodes:
+            if self._is_host_can_be_defined(csi_node.name):
+                self._add_node_to_nodes(csi_node)
+
+    def watch_csi_nodes_resources(self):
+        while True:
+            resource_version = self.csi_nodes_api.get().metadata.resourceVersion
+            stream = self.csi_nodes_api.watch(resource_version=resource_version, timeout=5)
+            for event in stream:
+                csi_node = self._get_csi_node_object(event[settings.OBJECT_KEY])
+                if (event[settings.TYPE_KEY] == settings.DELETED_EVENT) and (csi_node.name in NODES):
+                    self._handle_deleted_csi_node_pod(csi_node.name)
+                elif event[settings.TYPE_KEY] == settings.MODIFIED_EVENT:
+                    self._handle_modified_csi_node(csi_node)
+
+    def _handle_modified_csi_node(self, csi_node):
+        if csi_node.node_id and self._is_host_can_be_defined(csi_node.name) and csi_node.name not in NODES:
+            self._add_node_to_nodes(csi_node)
+            self._define_host_on_all_storages_from_secrets(csi_node.name)
+        elif csi_node.name in NODES:
+            self._handle_deleted_csi_node_pod(csi_node.name)
+
+    def _handle_deleted_csi_node_pod(self, node_name):
+        if self._is_host_can_be_undefined(node_name):
+            remove_host_thread = Thread(target=self._undefine_host_when_node_pod_is_deleted, args=(node_name,))
+            remove_host_thread.start()
+        else:
+            NODES.pop(node_name)
+
+    def _undefine_host_when_node_pod_is_deleted(self, node_name):
+        if self._is_host_part_of_update(node_name):
+            pass
+        else:
+            self._undefine_hosts(node_name)
+
+    def _is_host_part_of_update(self, worker):
+        daemon_set_name = self._wait_until_all_daemon_set_pods_are_up_to_date()
+        if daemon_set_name:
+            return self._is_csi_ibm_block_node_pod_running_on_worker(worker, daemon_set_name)
+        return False
+
+    def _wait_until_all_daemon_set_pods_are_up_to_date(self):
+        csi_ibm_block_daemon_set = self._get_csi_ibm_block_daemon_set()
+        if not csi_ibm_block_daemon_set:
+            return None
+        status = csi_ibm_block_daemon_set.status
+        while status.updated_number_scheduled != status.desired_number_scheduled:
+            if status.desired_number_scheduled == 0:
+                return None
+            csi_ibm_block_daemon_set = self._get_csi_ibm_block_daemon_set()
+            if not csi_ibm_block_daemon_set:
+                return None
+            status = csi_ibm_block_daemon_set.status
+            time.sleep(0.5)
+        return csi_ibm_block_daemon_set.metadata.name
+
+    def _is_csi_ibm_block_node_pod_running_on_worker(self, worker, daemon_set_name):
+        csi_ibm_block_pods = self._get_csi_ibm_block_pods()
```

Comment:
```suggestion
    def _is_csi_node_pod_running_on_worker(self, worker, daemon_set_name):
        csi_pods = self._get_csi_pods()
```
I think `ibm_block` could be implied throughout the project, so no need to include it explicitly almost everywhere